### PR TITLE
Reset splits either side of a divider to an even ratio on double click of the divider #2687

### PR DESCRIPTION
**Pull request by lesce**
This feat was requested here https://github.com/zeit/hyper/issues/2687
Should be ready to merge

### CODE REVIEW

File: lib/components/split-pane.js [hunk 1]
```diff
@@ -53,9 +75,13 @@ export default class SplitPane extends PureComponent {
...
       const count = new Array(total).fill(1 / total);
 
-      sizes = count;
       sizes_ = count;
     }
+    return sizes_;
+  }
+
+  onDrag(ev) {
+    let sizes_ = this.getSizes();
```
Comment (albinekb) [inline lib/components/split-pane.js]: Does this need to be run on every drag event?

Comment (timothyis) [reply lib/components/split-pane.js]: I guess it was already doing that. This just makes it accessible from elsewhere.

Comment (albinekb) [reply lib/components/split-pane.js]: Oh, never mind. Yep it was 👍

File: lib/components/split-pane.js [hunk 2]
```diff
@@ -18,6 +20,28 @@ export default class SplitPane extends PureComponent {
...
+    this.paneIndex = this.panes.indexOf(ev.target);
+    this.paneIndex -= Math.ceil(this.paneIndex / 2);
+  }
+
+  handleAutoResize(ev) {
+    ev.preventDefault();
+
+    this.setupPanes(ev);
+
+    let sizes_ = this.getSizes();
```
Comment (albinekb) [inline lib/components/split-pane.js]: `const`, you're not reassigning the reference later?

File: lib/components/split-pane.js [hunk 3]
```diff
@@ -18,6 +20,28 @@ export default class SplitPane extends PureComponent {
...
+    sizes_[this.paneIndex] = 0;
+    sizes_[this.paneIndex + 1] = 0;
+
+    let availableWidth = 1 - _.sum(sizes_);
```
Comment (albinekb) [inline lib/components/split-pane.js]: same here

File: lib/components/split-pane.js (hunk 1 above)
Comment (albinekb) [reply lib/components/split-pane.js]: `const` here too I think 😃

**Comment by lesce** [comment]
@albinekb I've addressed your feedback , thanks for the quick review 👍 💯 🥇

**Comment by timothyis** [comment]
Thank you 🙌

File: lib/components/split-pane.js [hunk 4]
```diff
@@ -18,6 +20,28 @@ export default class SplitPane extends PureComponent {
     }
   }
 
+  setupPanes(ev) {
```
Comment (chabou) [inline lib/components/split-pane.js]: This function deserves a better name. It doesn't setup anything. It saves dragging pane Index.

File: lib/components/split-pane.js [hunk 5]
```diff
@@ -18,6 +20,28 @@ export default class SplitPane extends PureComponent {
...
+    this.panes = Array.from(ev.target.parentNode.childNodes);
```
Comment (chabou) [inline lib/components/split-pane.js]: (This is not your code originally but..)
Why do not use a local variable ?

Btw, I think it is better to not determine index from DOM. Certainly better to add it to handler when rendering divider.

File: lib/components/split-pane.js [hunk 6]
```diff
@@ -18,6 +20,28 @@ export default class SplitPane extends PureComponent {
...
+    this.paneIndex = this.panes.indexOf(ev.target);
```
Comment (chabou) [inline lib/components/split-pane.js]: (This is not your code originally but..)
paneIndex is valide/set only after a dragStart.
Maybe better to have a name like : `this.draggingPaneIndex`

But if we use variable when rendering, we don't need it anymore

File: lib/components/split-pane.js [hunk 7]
```diff
@@ -18,6 +20,28 @@ export default class SplitPane extends PureComponent {
...
+  handleAutoResize(ev) {
+    ev.preventDefault();
+
+    this.setupPanes(ev);
+
+    const sizes_ = this.getSizes();
+    sizes_[this.paneIndex] = 0;
+    sizes_[this.paneIndex + 1] = 0;
+
+    const availableWidth = 1 - _.sum(sizes_);
```
Comment (chabou) [inline lib/components/split-pane.js]: I prefer using map/filter/reduce when it is possible :)

`const availableWidth = sizes_.reduce((total, size) => total - size, 1);`

**Comment by chabou** [comment]
@lesce Thank you so much for your work 🙏

**Pull request merged**